### PR TITLE
Add comments to the Dockerfile explaining how it all works

### PR DESCRIPTION
**Pull request by tas50**
Help outsiders understand how we push things up.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+#
+# When a build makes it through our internal CI system and is promoted to our "unstable" channel Expeditor will
+# trigger a Docker image build of that version and push it to Docker Hub.
+#
+# When tests of an unstable build pass within our CI system it will be promoted to the "current" channel and
+# Expeditor will tag that image as "current" on Docker Hub.
+#
+# When a build is promoted to our "stable" channel .expeditor/update_dockerfile.sh is run to update the version
+# in this file and also tag that image as "latest" and "stable" on Docker Hub. Additionally major and minor tags will be
+# applied so 15.0.260 would be tagged as "latest", "stable", "15" and "15.0", as well as "15.0.260".
```
Comment (tduffield) [inline Dockerfile]: We actually don't apply a "stable" tag, only "latest"

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,24 @@
+# About this Dockerfile:
+# Run without any arguments passed in this Dockerfile will build the latest "stable" channel release of Chef, which
+# is manually defined in the VERSION arg.
```
Comment (btm) [inline Dockerfile]: > Run without any arguments passed in this Dockerfile will build the latest "stable" channel release of Chef, which is manually defined in the VERSION arg.

How VERSION is updated is kinda buried. How about:

When run without any arguments passed, this Docker file will build the latest "stable" release of Chef. The version of that release is specified in the VERSION arg in this file, and is automatically updated as described below.

File: Dockerfile (hunk 1 above)
Comment (tas50) [reply Dockerfile]: Fixed. That was a dokken-ism

File: Dockerfile (hunk 2 above)
Comment (tas50) [reply Dockerfile]: Fixed

**Pull request merged**